File: Data Structures & Algorithms/string-encode-and-decode/submission-0.java
class Solution {

    public String encode(List<String> strs) {
        String result = "";

        for(String str : strs){
            result += "" + str.length() + "#" + str;
        }

        return result;
    }

    public List<String> decode(String str) {
        if(str.length() == 0) return new ArrayList<String>();

        List<String> result = new ArrayList<String>();

        int i = 0;

        while(i < str.length()){
            String num = ""; 

            while(str.charAt(i) != '#'){
                num += str.charAt(i);
                i++;
            }

            int val = Integer.parseInt(num) + i;

            i++;

            String s = "";

            while(i <= val){ 
                s += str.charAt(i);
                i++;
            }

            result.add(s);
        }

        return result;
    }
}

File: Data Structures & Algorithms/pow-x-n/submission-0.py
class Solution:
    def helper(self, x, n) -> float:
        if(n < 2):
            return x
        val = self.helper(x,n//2)
        val *= val
        if n%2 != 0:
            val *= x
        return val

    def myPow(self, x: float, n: int) -> float:
        if(x == 0):
            return 0
        if(n==0):
            return 1
        
        b = abs(n)
        result = self.helper(x, b)

        
        if n < 0:
            return 1 / result
        return result
